lisa/sut_orchestrator/azure/arm_template.bicep

```diff
@@ -0,0 +1,349 @@
+  version: node.marketplace.version
+}
+
+func getSecurityProfileForOSDisk(node object) object => empty(node.security_profile.disk_encryption_set_id)
+? {
+  securityEncryptionType: node.security_profile.encryption_type
+} : {
+  securityEncryptionType: node.security_profile.encryption_type
+  diskEncryptionSet: {
+    id: node.security_profile.disk_encryption_set_id
+  }
+}
+
+func generateOsProfile(node object, admin_username string, admin_password string, admin_key_data string) object => {
+  computername: node.short_name
+  adminUsername: admin_username
+  adminPassword: (empty(admin_password) ? null : admin_password)
+  linuxConfiguration: (((!empty(admin_key_data)) && node.is_linux) ? getLinuxConfiguration('/home/${admin_username}/.ssh/authorized_keys', admin_key_data, empty(admin_password)) : null)
+}
+
+func generateImageReference(node object) object => (isVhd(node) ? getOsDiskVhd(node.name)
+: ((!empty(node.shared_gallery))
+? getOsDiskSharedGallery(node.shared_gallery)
+: getOsDiskMarketplace(node)))
+
+func generateSecurityProfile(node object) object => {
+  uefiSettings: {
+    secureBootEnabled: node.security_profile.secure_boot
+    vTpmEnabled: true
+  }
+  securityType: node.security_profile.security_type
+}
+
+func getOsProfile(node object, admin_username string, admin_password string, admin_key_data string) object => isCvm(node) 
+? {}
+: generateOsProfile(node, admin_username, admin_password, admin_key_data)
+
+func getImageReference(node object) object => isCvm(node) 
+? {}
+: generateImageReference(node)
+
+func getSecurityProfile(node object) object => empty(node.security_profile) 
+? {} 
+: generateSecurityProfile(node)
+
+func getOSImage(node object) object => {
+  name: '${node.name}-osDisk'
+  managedDisk: {
+    storageAccountType: node.os_disk_type
+    securityProfile: (empty(node.security_profile) || (node.security_profile.security_type != 'ConfidentialVM')) ? null : getSecurityProfileForOSDisk(node)
+  }
+  caching: ((node.os_disk_type == 'Ephemeral') ? 'ReadOnly' : 'ReadWrite')
+  createOption: 'FromImage'
+  diskSizeGB: node.osdisk_size_in_gb
+}
+
+
+func getAvailabilitySetId(availability_set_name string) object => {
+  id: resourceId('Microsoft.Compute/availabilitySets', availability_set_name)
+}
+
+func isCvm(node object) bool => bool((!empty(node.vhd)) && (!empty(node.vhd.vmgs_path)))
+
+func isVhd(node object) bool => bool((!empty(node.vhd)) && (!empty(node.vhd.vhd_path)))
+
+func getVMOsDisk(node object) object => isCvm(node) ? getOSDisk('${node.name}-disk')
+: ((node.os_disk_type == 'Ephemeral')
+? getEphemeralOSImage(node)
+: getOSImage(node))
+
+func getDataDisk(nodeName string, dataDisk object, index int) object => (dataDisk.type == 'UltraSSD_LRS')
+? getAttachDisk(dataDisk, '${nodeName}-data-disk-${index}', index)
+: getCreateDisk(dataDisk, '${nodeName}-data-disk-${index}', index)
+
+resource nodes_vms 'Microsoft.Compute/virtualMachines@2022-08-01' = [for i in range(0, node_count): {
```

Comment:
Move resources together. Don't mix func and resources.

Reply:
done

Reply:
done